game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -1202,6 +1162,86 @@ && canAirSafelyLandAfterAttack(unit, t)) {
     }
   }
 
+  private void determineTerritoriesThatCanBeBombed(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Unit> alreadyAttackedWithUnits) {
+    final Map<Unit, Set<Territory>> bomberMoveMap =
+        territoryManager.getAttackOptions().getBomberMoveMap();
+    final Set<Unit> bombers = bomberMoveMap.keySet();
+    bombers.removeAll(alreadyAttackedWithUnits);
+    final boolean raidsMayBePrecededByAirBattles =
+        Properties.getRaidsMayBePreceededByAirBattles(data.getProperties());
+    for (final Unit bomber : bombers) {
+      final Set<Territory> bomberTargetTerritories = bomberMoveMap.get(bomber);
+      bomberTargetTerritories.removeIf(
+          terr ->
+              raidsMayBePrecededByAirBattles
+                  && AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(
+                      terr, player, data, true));
+      determineBestBombingAttackForBomber(
+          attackMap, sortedUnitAttackOptions, bomberTargetTerritories, bomber);
+    }
+  }
+
+  /**
+   * Determines the best bombing attack territory for a single {@code bomber}.
+   *
+   * @param attackMap Attack map
+   * @param sortedUnitAttackOptions Sorted attack options
+   * @param bomberTargetTerritories Target territories
+   * @param bomber Bomber unit
+   */
+  private void determineBestBombingAttackForBomber(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Territory> bomberTargetTerritories,
+      final Unit bomber) {
+    final Predicate<Unit> bombingTargetMatch =
+        Matches.unitCanProduceUnitsAndCanBeDamaged()
+            .and(Matches.unitIsLegalBombingTargetBy(bomber));
+    Optional<Territory> maxBombingTerritory = Optional.empty();
+    int maxBombingScore = MIN_BOMBING_SCORE;
+    for (final Territory t : bomberTargetTerritories) {
+      final List<Unit> targetUnits = t.getUnitCollection().getMatches(bombingTargetMatch);
+      if (!targetUnits.isEmpty() && canAirSafelyLandAfterAttack(bomber, t)) {
+        final int noAaBombingDefense =
+            t.anyUnitsMatch(Matches.unitIsAaForBombingThisUnitOnly()) ? 0 : 1;
+        int maxDamageProduction = 0; // territory damage + target unit damage(s)
+        final TerritoryAttachment ta = TerritoryAttachment.get(t);
+        if (ta != null) {
+          maxDamageProduction = ta.getProduction();
+        }
+        int neededDamageUnits = 0;
+        final List<Unit> existingAttackingBombers = attackMap.get(t).getBombers();
+        for (Unit target : targetUnits) {
+          neededDamageUnits += target.getHowMuchDamageCanThisUnitTakeTotal(t);
```

Comment:
Interesting. Looking at its implementation, I see it has a check for:

Properties.getDamageFromBombingDoneToUnitsInsteadOfTerritories()

That seems to imply that if the property is set, then we shouldn't be using ta.getProduction().

We should check with a rules expert, but that seems to imply that the max damage should be either:

if (Properties.getDamageFromBombingDoneToUnitsInsteadOfTerritories()) {
  max = sum(target.getHowMuchDamageCanThisUnitTakeTotal(t));
} else {
  max = TerritoryAttachment.getProduction(t)
}

Your logic doesn't seem to match that. But maybe I'm missing something.

Reply:
I think we are not 100% clear about the difference of damage to the unit or the underlying production of this unit.
From my understanding the method `getHowMuch**More**DamageCanThisUnitTake` should be used.

However, it seemed to me that there was a mistake and I would like to put my thoughts out on the to-be logic which you can comment about separate from the (modified) code now:
`remainingDamagePotential = 

1: maxDamageProduction (of the territory) 
2: - neededDamageUnits (= damage still possible of all units  that can be bombed) 
3. - 3 * numExistingBombers (assumed to be attacking together with the current bomber)`
Note that 3. leave the potential that not all (co-)bombers assumed to attack together might join based on the individual target assessment run. Furthermore, each co-bomber should be counted only ones.